### PR TITLE
Clang14 and CUDA11.2 in AFQMC-Offload CI

### LINKED ISSUE #3953: Adjust offload CI needed
**Is your feature request related to a problem? Please describe.**
Due to #3922,  #3921 was failing CI.
This issue was not exposed due to the fact that CI didn't test complex case.

**Describe the solution you'd like**
Need to force CUDA 11.2 instead of a version found under /usr/local/cuda

### LINKED ISSUE #3960: Bump CI LLVM to v14
**Is your feature request related to a problem? Please describe.**
We are currently testing offload functionality on CI using a LLVM development branch prior to v14

**Describe the solution you'd like**
Replace current LLVM build with v14.0.1 

**Describe alternatives you've considered**
None

**Additional context**
Development branch was needed due to recently bug fixes in LLVM

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

Bump to Clang14  in AFQMC-Offload CI test from using a non-release commit
Fixes #3960
Add Complex tests for CUDA 11.2. 
Fixes #3953 
Complex test are broken with CUDA 11.6
Must be merged to work

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Bugfix
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
sulfur, must pass CI

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: tests/test_automation/github-actions/ci/run_step.sh [hunk 1]
```diff
@@ -33,11 +33,23 @@ case "$1" in
 
     if [[ "${GH_JOBNAME}" =~ (-CUDA) ]]
     then
-      echo "Set CUDACXX CMake environment variable to nvcc standard location"
-      export CUDACXX=/usr/local/cuda/bin/nvcc
-        
-      # Make current environment variables available to subsequent steps
-      echo "CUDACXX=/usr/local/cuda/bin/nvcc" >> $GITHUB_ENV
+      
+      if [[ "${GH_JOBNAME}" =~ (-AFQMC-Offload-) ]]
```
Comment (prckent) [inline tests/test_automation/github-actions/ci/run_step.sh:37]: Not critical to change, but I think we only need Offload here. A no AFQMC build would also need this fix.

Comment (williamfgc) [reply tests/test_automation/github-actions/ci/run_step.sh:37]: Good point, we have a `Clang12-NoMPI-Offload-Real` job in GitHub Actions hosted runners. The AFQMC-Offload makes it less invasive with the previos, but it's good to keep this things in mind as CI evolves.

Comment (ye-luo) [reply tests/test_automation/github-actions/ci/run_step.sh:37]: I think `Clang12-NoMPI-Offload-Real` was for the host. We might better to rename it as `Clang12-NoMPI-Offload-Host-Real`

File: tests/test_automation/github-actions/ci/run_step.sh [hunk 2]
```diff
@@ -320,6 +325,8 @@ case "$1" in
     then
       if [[ "${GH_JOBNAME}" =~ (-AFQMC-Offload-) ]]
       then
+        # adding PATH to prevent default /usr/local/cuda to be associated with the compiler
+        export PATH=/usr/local/cuda-11.2/bin:$PATH
```
Comment (ye-luo) [inline tests/test_automation/github-actions/ci/run_step.sh:329]: Test phase doesn't need path

File: tests/test_automation/github-actions/ci/run_step.sh (hunk 1 above)
Comment (ye-luo) [reply tests/test_automation/github-actions/ci/run_step.sh:37]: Please also remove AFQMC. Thanks to the outer if on `-CUDA`, offload to host won't be affected.

**Review by ye-luo — CHANGES_REQUESTED** [review]

**Review by ye-luo — APPROVED** [review]
Keep things moving.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Clang14Dev-MPI-CUDA-AFQMC-Offload-XXX failure is expected as we changed the job name. I temporarily removed the protection and will add them back with new names.

**Comment by williamfgc** [comment]
Thanks @ye-luo

**Pull request merged**